tools/cldr-code/src/main/java/org/unicode/cldr/util/SupplementalCalendarData.java

```diff
@@ -0,0 +1,229 @@
+package org.unicode.cldr.util;
+
+import com.ibm.icu.impl.locale.XCldrStub.ImmutableMap;
+import com.ibm.icu.util.GregorianCalendar;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.function.Consumer;
+import java.util.function.Supplier;
+import org.unicode.cldr.icu.LDMLConstants;
+
+public class SupplementalCalendarData implements Iterable<String> {
+    /** an <era> element */
+    public static class EraData implements Comparable<EraData> {
+        static final int INDEX = 4; // index of our element
+
+        /** the <era> xpath */
+        private final XPathValue xpath;
+
+        EraData(final XPathValue xpath) {
+            this.xpath = xpath;
```

Comment:
In general, for sup. data we have parsed out the data from each path and cached it in a friendly data structure, rather than parsing on every single call to access data.

Reply:
If you don't have time to do those changes now, please file a ticket to fix in the future.

Reply:
It actually doesn't re parse, the only overhead is a hash table look up on the frozen Xpath.

Reply:
I was trying to keep the parser  simple as a model for others that need to be added

Reply:
done